packages/modelence/src/cron/jobs.ts

```diff
@@ -66,106 +60,104 @@ export function defineCronJob(
-          }
-        )
-      )
-    );
+  if (aliasList.length === 0) {
+    return;
+  }
 
-    if (existingLockedRecord) {
-      await sleep(LOCK_TRANSFER_DELAY);
+  cronJobsStarted = true;
+
+  await tryAcquireLockAndStartJobs();
+
+  setInterval(async () => {
+    if (!cronJobsInterval) {
+      // We don't have the lock, try to acquire it
+      await tryAcquireLockAndStartJobs();
     }
+  }, LOCK_RETRY_INTERVAL);
+}
 
-    const cronJobRecords = await cronJobsCollection.fetch(aliasSelector);
-    const now = Date.now();
-    cronJobRecords.forEach((record) => {
-      const job = cronJobs[record.alias];
-      if (!job) {
-        return;
-      }
-      job.state.scheduledRunTs = record.lastStartDate
-        ? record.lastStartDate.getTime() + job.params.interval
-        : now;
-    });
-    Object.values(cronJobs).forEach((job) => {
-      if (!job.state.scheduledRunTs) {
-        job.state.scheduledRunTs = now;
-      }
-    });
+/**
+ * Attempts to acquire the cron lock and start the cron job scheduler.
+ * If the lock is acquired, initializes job schedules and starts the tick interval.
+ */
+async function tryAcquireLockAndStartJobs() {
+  const aliasList = Object.keys(cronJobs);
+  if (aliasList.length === 0) {
+    return;
+  }
 
-    cronJobsInterval = setInterval(tickCronJobs, time.seconds(1));
+  // Try to acquire the cron lock
+  const lockAcquired = await acquireLock('cron');
+
+  if (!lockAcquired) {
+    return;
   }
-}
 
-function sleep(ms: number) {
-  return new Promise((resolve) => setTimeout(resolve, ms));
+  // Load job schedules from database
+  const cronJobRecords = await cronJobsCollection.fetch({
+    alias: { $in: aliasList },
+  });
+  const nowTimestamp = Date.now();
+  cronJobRecords.forEach((record) => {
+    const job = cronJobs[record.alias];
+    if (!job) {
+      return;
+    }
+    job.state.scheduledRunTs = record.lastStartDate
+      ? record.lastStartDate.getTime() + job.params.interval
+      : nowTimestamp;
+  });
+  Object.values(cronJobs).forEach((job) => {
+    if (!job.state.scheduledRunTs) {
+      job.state.scheduledRunTs = nowTimestamp;
+    }
+  });
+
+  // Start the cron job tick interval and heartbeat
+  cronJobsInterval = setInterval(tickCronJobs, time.seconds(1));
 }
 
 async function tickCronJobs() {
-  // TODO: periodically check if the locks are still there
-
   const now = Date.now();
-  Object.values(cronJobs).forEach(async (job) => {
+  const jobs = Object.values(cronJobs);
+
+  for (const job of jobs) {
     const { params, state } = job;
     if (state.isRunning) {
       if (state.startTs && state.startTs + params.timeout < now) {
         // TODO: log cron trace timeout error
         state.isRunning = false;
       }
-      return;
+      continue;
     }
 
     // TODO: limit the number of jobs running concurrently
 
     if (state.scheduledRunTs && state.scheduledRunTs <= now) {
-      await startCronJob(job);
+      const ownsLock = await acquireLock('cron');
```

Comment:
Why do we need this check for each cron job? I think we just need to re-acquire the lock at the beginning of the tick function if it's time for the next "heartbeat", and if it's successful there are no other checks needed?